### PR TITLE
Demonstrate using PyStow for downloading S3 SQL content

### PR DESCRIPTION
**Pull request by cthoyt**
Related to #136 

PyStow has all of the logic of caching files from a URL (including deterministically naming the files) built in. It can be used to replace a small amount of code that works with S3 to get pre-build SQLite versions of ontologies.

- [x] Replaces implementation of `download()`
- [x] Replaces downloading logic with PyStow's `ensure()` functionality 
- [x] Documents how PyStow chooses its caching location (e.g., it puts stuff in `~/.data/oaklib` but this can be modified with certain environment variables to either use XDG-compliant paths or to change `.data` to something else explicitly)

I've also given a second option in the comments that uses PyStow's S3 API wrapper (via `botocore`)

### CODE REVIEW

File: src/oaklib/implementations/sqldb/sql_implementation.py [hunk 1]
```diff
@@ -95,6 +90,12 @@
 from oaklib.interfaces.validator_interface import ValidatorInterface
 from oaklib.types import CURIE, SUBSET_CURIE
 
+__all__ = [
```
Comment (cmungall) [inline src/oaklib/implementations/sqldb/sql_implementation.py:93]: Aside: Do you think we should implement this pattern throughout

(when looking at this I wonder if these should actually be not exported and start with `_` but can handle that separately)

File: src/oaklib/implementations/sqldb/sql_implementation.py [hunk 2]
```diff
@@ -111,14 +112,6 @@ def _mapping(m: sssom.Mapping):
     return m
 
 
-# https://stackoverflow.com/questions/16694907/download-large-file-in-python-with-requests
```
Comment (cmungall) [inline src/oaklib/implementations/sqldb/sql_implementation.py:114]: glad to see this go.

Note to self: we have gz files on s3 now will want to download these preferentially

**Review by cmungall — APPROVED** [review]

**Comment by cmungall** [comment]
We discussed this a bit externally, there was some concern about introducing an additional dependency for this task, but I pointed out we are already using appdirs

There was also some concern about hardwiring to S3, but that had nothing to do with this PR (the option to use S3-specific is commented out), I made a separate issue to consider PURLs for the sqlite downloads: https://github.com/INCATools/semantic-sql/issues/43

File: src/oaklib/implementations/sqldb/sql_implementation.py (hunk 1 above)
Comment (cthoyt) [reply src/oaklib/implementations/sqldb/sql_implementation.py:93]: Yes, we should carefully do exports everywhere.

**Pull request merged**